src/radical/pilot/agent/agent_0.py

```diff
@@ -340,77 +353,130 @@ def _start_services(self):
         If a `./services` file exist, reserve a compute node and run that file
         there as bash script.
         '''
-
-        if not os.path.isfile('./services'):
-            return
+        service_descriptions = self._cfg.services
+        services = list()
+        self._service_task_ids = {}
+        for service_desc in service_descriptions:
+            task = dict()
+            task['origin'] = 'services'
+            task['description'] = TaskDescription(service_desc).as_dict()
+            task['state'] = rps.AGENT_STAGING_INPUT_PENDING
+            task['status'] = 'NEW'
+            task['type'] = 'task'
+            task['uid'] = ru.generate_id('services.%(item_counter)04d',
+                                             ru.ID_CUSTOM, ns="TODO_WHATTOUSE")
+            self._service_task_ids.add(task['uid'])
+            task['task_sandbox_path'] = self._sbox
+            task['task_sandbox'] = 'file://localhost/' + self._sbox
+            task['pilot_sandbox'] = self._cfg.pilot_sandbox
+            # task['session_sandbox'] = os.environ['RP_SESSION_SANDBOX']
+            # task['resource_sandbox'] = os.environ['RP_RESOURCE_SANDBOX']
+            task['pilot'] = self._cfg.pid
+            # task['resources'] = {'cpu': td['ranks'] *
+            #                             td.get('cores_per_rank', 1),
+            #                      'gpu': td['ranks'] *
+            #                             td.get('gpus_per_rank', 1)}
+            services.append(task)
+        self.number_of_services_to_launch = len(services)
+        self.advance(services, publish=False, push=True)
+        # Waiting 2mins for all services to launch
+        
+        did_timed_out = self.services_event.wait(timeout=60*2)
+        if not did_timed_out:
+            raise Exception("Unable to start services") # #TODO custom exception
+
+        # if not os.path.isfile('./services'):
+        #     return
 
         # launch the `./services` script on the service node reserved by the RM.
-        nodes = self._rm.info.service_node_list
-        assert nodes
-
-        bs_name = "%s/bootstrap_2.sh"     % self._pwd
-        ls_name = "%s/services_launch.sh" % self._pwd
-        ex_name = "%s/services_exec.sh"   % self._pwd
-
-        service_task_uid = 'rp.services'
-        service_task     = {
-            'uid'               : service_task_uid,
```

Comment:
All this code shall not be used @andre-merzky  I think. do you see the same

Reply:
right

Reply:
ok